corehq/apps/hqcase/case_helper.py

```diff
@@ -0,0 +1,188 @@
+                f'{domain!r}')
+
+            self.case = case
+        elif case_id:
+            self.case = BriefCase(case_id=case_id, domain=domain)
+        else:
+            self.case = None
+        self.domain = domain
+
+    def create_case(
+        self,
+        case_data,
+        *,
+        user_id=None,
+        device_id='corehq.apps.hqcase.case_helper.CaseHelper',
+    ):
+        assert self.case is None, '`self.case` already exists'
+
+        case_data = self._copy_case_data(case_data)
+        user = self._get_user_duck(user_id, self.domain)
+        case_data.setdefault('user_id', user.user_id)
+        case_data.setdefault('owner_id', '')
+        __, self.case = handle_case_update(
+            self.domain,
+            case_data,
+            user=user,
+            device_id=device_id,
+            is_creation=True,
+        )
+
+    def update(
+        self,
+        case_data,
+        *,
+        user_id=None,
+        device_id='corehq.apps.hqcase.case_helper.CaseHelper',
+    ):
+        if not case_data:
+            return
+
+        assert self.case, '`self.case` is not defined'
+
+        case_data = self._copy_case_data(case_data)
+        case_data['case_id'] = self.case.case_id
+        user = self._get_user_duck(user_id, self.domain)
+        case_data.setdefault('user_id', user.user_id)
+        __, self.case = handle_case_update(
+            self.domain,
+            case_data,
+            user=user,
+            device_id=device_id,
+            is_creation=False,
+        )
+
+    def close(
+        self,
+        *,
+        user_id=None,
+        device_id='corehq.apps.hqcase.case_helper.CaseHelper',
+    ):
+        assert self.case, '`self.case` is not defined'
+
+        user = self._get_user_duck(user_id, self.domain)
+        case_data = {
+            'case_id': self.case.case_id,
+            'user_id': self.case.user_id or user.user_id,
+            'close': True,
+        }
+        __, self.case = handle_case_update(
+            self.domain,
+            case_data,
+            user=self._get_user_duck(user_id, self.domain),
+            device_id=device_id,
+            is_creation=False,
+        )
+
+    @staticmethod
+    def _copy_case_data(case_data):
+        """
+        Given the output of Case API v0.6's ``serialize_case()``,
+        returns a new dictionary with invalid fields omitted.
+
+        >>> case_data = {'case_id': 'abc123', 'domain': 'test', 'foo': 'bar'}
+        >>> CaseHelper._copy_case_data(case_data)
+        {'foo': 'bar'}
+
+        """
+        invalid_fields = {
+            'case_id',
+            'domain',
+            'closed',
+            'date_closed',
+            'date_opened',
+            'last_modified',
+            'modified_by',
+            'server_last_modified',
+            'indexed_on',
+        }
+        return {k: v for k, v in case_data.items() if k not in invalid_fields}
```

Comment:
I think this method should raise an error rather than ignoring invalid items because it would surface errors sooner (rather than allowing them to pass silently).

Reply:
If we drop this method, `handle_case_update()` will raise the validation errors we would want.

I want `CaseHelper.create_case()` to be able to accept the output of `serialize_case()`, without the user needing to know how to clean up the data first.

I am looking for a standard format for a case. Initially I thought the output of `CommCareCase.to_api_json()` could be that format, but the Case API v0.6 uses a different format. So I'm hoping that the output of `serialize_case()` can be that format, and we can pass it straight to `CaseHelper.create_case()` without a problem.

Reply:
> If we drop this method, `handle_case_update()` will raise the validation errors we would want.

This sounds like a good simplification.

> I want `CaseHelper.create_case()` to be able to accept the output of `serialize_case()`, without the user needing to know how to clean up the data first.

This also sounds convenient.

> I am looking for a standard format for a case...

Sounds like there is an unsolved problem here yet. I'm hoping this was mostly you thinking out loud because I'm not sure what to suggest without digging deeper into the code.

Reply:
> This sounds like a good simplification. ... This also sounds convenient.

I think we can have both, if we raise validation errors by default, and require the user to ask explicitly for the data to be cleaned to avoid the validation errors: dfc4cfba976

Updated tests show that validation errors are raised unless the user passes `is_serialized=True` to avoid them.